pandora-server-directory/src/database/mongoDb.ts

```diff
@@ -593,6 +608,79 @@ export default class MongoDatabase implements PandoraDatabase {
 
 	private async doManualMigrations(): Promise<void> {
 		// Add manual migrations here
+
+		//#region Migrate character self info from accounts to characters
+		const charactersToMigrate = new Map<CharacterId, { account: AccountId; character: DatabaseCharacterSelfInfo; }>();
+
+		// Gather data about characters
+		await accountCollection.doManualMigration(this._client, this._db, {
+			oldSchema: DatabaseAccountWithSecureSchema.extend({
+				characters: DatabaseCharacterSelfInfoSchema
+					.omit({ currentSpace: true })
+					.and(z.object({ currentRoom: SpaceIdSchema.nullable().optional() }))
+					.array()
+					.optional(),
+			}),
+			migrate: async ({ oldStream }) => {
+				for await (const account of oldStream) {
+					if (Array.isArray(account?.characters)) {
+						for (const character of account.characters) {
+							Assert(!charactersToMigrate.has(character.id));
+
+							charactersToMigrate.set(character.id, {
+								account: account.id,
+								character: {
+									id: character.id,
```

Comment:
account.character will be removed by automatic migration right?

Reply:
Yes, if the migration does run in "migrate" mode. I will look into making that automatic if needed~